MapboxNavigation/NavigationMapView.swift

```diff
@@ -885,64 +870,57 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
     }
     
     func routeWaypointSymbolStyleLayer(identifier: String, source: MGLSource) -> MGLStyleLayer {
         let symbol = MGLSymbolStyleLayer(identifier: identifier, source: source)
         
-        symbol.text = MGLStyleValue(rawValue: "{name}")
-        symbol.textFontSize = MGLStyleValue(rawValue: 10)
-        symbol.textHaloWidth = MGLStyleValue(rawValue: 0.25)
-        symbol.textHaloColor = MGLStyleValue(rawValue: .black)
+        symbol.text = NSExpression(forConstantValue: "{name}")
+        symbol.textFontSize = NSExpression(forConstantValue: 10)
+        symbol.textHaloWidth = NSExpression(forConstantValue: 0.25)
+        symbol.textHaloColor = NSExpression(forConstantValue: UIColor.black)
         
         return symbol
     }
     
     func routeStyleLayer(identifier: String, source: MGLSource) -> MGLStyleLayer {
         
         let line = MGLLineStyleLayer(identifier: identifier, source: source)
-        line.lineWidth = MGLStyleValue(interpolationMode: .exponential,
-                                       cameraStops: routeLineWidthAtZoomLevels,
-                                       options: [.defaultValue : MGLConstantStyleValue<NSNumber>(rawValue: 1.5)])
-        
-        line.lineColor = MGLStyleValue(interpolationMode: .categorical, sourceStops: [
-            "unknown": MGLStyleValue(rawValue: trafficUnknownColor),
-            "low": MGLStyleValue(rawValue: trafficLowColor),
-            "moderate": MGLStyleValue(rawValue: trafficModerateColor),
-            "heavy": MGLStyleValue(rawValue: trafficHeavyColor),
-            "severe": MGLStyleValue(rawValue: trafficSevereColor)
-            ], attributeName: "congestion", options: [.defaultValue: MGLStyleValue(rawValue: trafficUnknownColor)])
-        
-        line.lineOpacity = MGLStyleValue(interpolationMode: .categorical, sourceStops: [
-            true: MGLStyleValue(rawValue: 1),
-            false: MGLStyleValue(rawValue: 0)
-            ], attributeName: currentLegAttribute, options: nil)
-        
-        line.lineJoin = MGLStyleValue(rawValue: NSValue(mglLineJoin: .round))
+        line.lineWidth = NSExpression(format: "FUNCTION($zoomLevel, 'mgl_interpolateWithCurveType:parameters:stops:', 'linear', nil, %@)", argumentArray: [routeLineWidthAtZoomLevels])
+        line.lineColor = NSExpression(format: "FUNCTION($congestion, 'mgl_stepWithMinimum:stops:', %@, %@)", UIColor(red:1.0, green:0.72, blue:0.85, alpha:1.0), [
+            "unknown": NSExpression(forConstantValue: trafficUnknownColor),
+            "low": NSExpression(forConstantValue: trafficLowColor),
+            "moderate": NSExpression(forConstantValue: trafficModerateColor),
+            "heavy": NSExpression(forConstantValue: trafficHeavyColor),
+            "severe": NSExpression(forConstantValue: trafficSevereColor)
+            ])
+        line.lineOpacity = NSExpression(format: "FUNCTION($currentLegAttribute, 'mgl_interpolateWithCurveType:parameters:stops:', 'linear', nil, %@)", [
```

Comment:
`$currentLegAttribute` would be a reference to an expression variable declared with `mgl_expressionWithContext:`. Use `currentLegAttribute`, without the dollar sign, to refer to an attribute of the current feature.

Reply:
Actually, `currentLegAttribute` is the name of [this variable](https://github.com/mapbox/mapbox-navigation-ios/blob/e942a425208feda1ad2a15ee0c6ad27552440b34/MapboxNavigation/NavigationMapView.swift#L62). You have two options: either insert a key path expression created from the variable:

```swift
NSExpression(format: "FUNCTION(%@, 'mgl_interpolateWithCurveType:parameters:stops:', 'linear', nil, %@)", NSExpression(forKeyPath: currentLegAttribute), …)
```

or inline the key path and delete `currentLegAttribute`:

```swift
NSExpression(format: "FUNCTION(isCurrentLeg, 'mgl_interpolateWithCurveType:parameters:stops:', 'linear', nil, %@)", …)
```